File: src/main/java/org/dukcode/ps/codetree/trail02/chapter04/lesson04/ChallengeTheTotalAreaOfColoredPaper.java
package org.dukcode.ps.codetree.trail02.chapter04.lesson04;

import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.IOException;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;
import java.util.Arrays;
import java.util.StringTokenizer;

/**
 * @see <a
 * href="https://www.codetree.ai/ko/trails/complete/curated-cards/challenge-the-total-area-of-colored-paper/description">LINK</a>
 */
public class ChallengeTheTotalAreaOfColoredPaper {

  private static final int RANGE = 100;

  private static BufferedReader br;
  private static BufferedWriter bw;
  private static StringTokenizer st;

  private static int[][] board;

  public static void main(String[] args) throws IOException {
    br = new BufferedReader(new InputStreamReader(System.in));
    bw = new BufferedWriter(new OutputStreamWriter(System.out));

    board = new int[2 * RANGE + 1][2 * RANGE + 1];

    int n = Integer.parseInt(br.readLine());
    for (int i = 0; i < n; i++) {
      st = new StringTokenizer(br.readLine());
      int fx = Integer.parseInt(st.nextToken());
      int fy = Integer.parseInt(st.nextToken());

      for (int y = fy; y < fy + 8; y++) {
        for (int x = fx; x < fx + 8; x++) {
          board[y + RANGE][x + RANGE] = 1;
        }
      }
    }

    int area = 0;
    for (int[] line : board) {
      area += Arrays.stream(line).sum();
    }

    bw.write(String.valueOf(area));

    br.close();
    bw.close();

  }
}

File: src/main/kotlin/org/dukcode/ps/codetree/trail02/chapter04/lesson04/ChallengeTheTotalAreaOfColoredPaper.kt
package org.dukcode.ps.codetree.trail02.chapter04.lesson04.challengeTheTotalAreaOfColoredPaper

import java.io.BufferedReader
import java.io.BufferedWriter
import java.io.InputStreamReader
import java.io.OutputStreamWriter
import java.util.StringTokenizer

private val br = BufferedReader(InputStreamReader(System.`in`))
private val bw = BufferedWriter(OutputStreamWriter(System.out))

private const val OFFSET = 100
private const val PAPER_SIZE = 8

/**
 * @see <a href="https://www.codetree.ai/ko/trails/complete/curated-cards/challenge-the-total-area-of-colored-paper/description">LINK</a>
 */
fun main() {
    val board =
        Array(2 * OFFSET + 1) {
            IntArray(2 * OFFSET + 1)
        }

    val n = br.readLine().toInt()
    repeat(n) {
        val (x1, y1) = parseInts()
        for (y in y1 until y1 + PAPER_SIZE) {
            for (x in x1 until x1 + PAPER_SIZE) {
                board[y + OFFSET][x + OFFSET] = 1
            }
        }
    }

    bw.write(board.sumOf(IntArray::sum).toString())

    bw.flush()
}

private fun parseInts() =
    StringTokenizer(br.readLine()).run {
        IntArray(countTokens()) {
            nextToken().toInt()
        }
    }
